src/celeritas/optical/ScintillationParams.cc

```diff
@@ -47,19 +47,22 @@ ScintillationParams::from_import(ImportData const& data,
     {
         // Collect ScintillationParticleIds
         input.pid_to_scintpid.resize(data.particles.size());
-        ScintillationParticleId scintpid;
+        ScintillationParticleId scintpid{0};
         for (auto const& [matid, iom] : data.optical)
         {
             auto const& iomsp = iom.scintillation.particles;
             for (auto const& [pdg, ipss] : iomsp)
             {
-                auto const pid = particle_params->find(PDGNumber{pdg});
-                // Add new ScintillationParticleId
-                input.pid_to_scintpid[pid.get()] = scintpid++;
+                if (auto const pid = particle_params->find(PDGNumber{pdg}))
+                {
+                    // Add new ScintillationParticleId
+                    input.pid_to_scintpid[pid.get()] = scintpid++;
+                }
             }
         }
         // Resize particle- and material-dependent spectra
-        input.particles.resize(input.pid_to_scintpid.size() * num_optmats);
+        auto const num_scint_particles = scintpid.get() + 1;
```

Comment:
```suggestion
        auto const num_scint_particles = scintpid.get();
```
Since `scintpid` is post-incremented.